toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/codedvalue/ComboBoxField.kt

```diff
@@ -281,51 +285,87 @@ internal fun ComboBoxDialog(
-                                        fontStyle = FontStyle.Italic,
-                                        fontWeight = FontWeight.Light
-                                    )
-                                    else LocalTextStyle.current
+                                    text = stringResource(R.string.unsupported_type),
+                                    style = MaterialTheme.typography.labelMedium
+                                )
+                                ComboBoxListItem(
+                                    name = initialValue.toString(),
+                                    isChecked = true,
+                                    textStyle = LocalTextStyle.current.copy(fontStyle = FontStyle.Italic),
+                                    onClick = {}
                                 )
-                            },
-                            modifier = Modifier
-                                .fillMaxWidth()
-                                .clickable {
-                                    // if the no value label was selected, set the value to be empty
-                                    if (name == noValueLabel) {
-                                        onValueChange(null)
-                                    } else {
-                                        onValueChange(code)
-                                    }
-                                }
-                                .semantics {
-                                    contentDescription = if (name == noValueLabel) {
-                                        "no value row"
-                                    } else {
-                                        "$name list item"
-                                    }
-                                },
-                            trailingContent = {
-                                if ((code == initialValue) || ((name == noValueLabel) && (initialValue == null))) {
-                                    Icon(
-                                        imageVector = Icons.Outlined.Check,
-                                        contentDescription = "list item check"
-                                    )
-                                }
                             }
-                        )
+                        }
                     }
                 }
             }
         }
     }
 }
 
-@Preview
+@Composable
+private fun ComboBoxListItem(
+    name: String,
+    isChecked: Boolean,
+    modifier: Modifier = Modifier,
+    textStyle: TextStyle = LocalTextStyle.current,
+    onClick: () -> Unit
+) {
+    ListItem(
+        headlineContent = {
+            Text(
+                text = name,
+                style = textStyle
+            )
+        },
+        modifier = modifier
+            .fillMaxWidth()
+            .clickable {
+                onClick()
+            },
+        trailingContent = {
+            if (isChecked) {
```

Comment:
the content could be null if not checked, as opposed to an empty composable.

```suggestion
        trailingContent = ifChecked {
            { ... }
        } else { 
            null
        }
```

Reply:
I believe it can be null. Looking at its signature -

```kotlin
public fun ListItem(
  ...
  trailingContent: @Composable() (() -> Unit)? = null
  ...
)
```